### DISCUSSION TITLE
Field Curvature, Distortion, Spot and Chromatic Aberration

### DISCUSSION THREAD

**Discussion by gniliac1** [mjhoptics/ray-optics#136] [Q&A] [answered]
Is there a possibility to calculate the sagittal/tangential field curvature and the maximal distortion (not only for third order?). I was searching for it but can't find it. As well as the RMS-Spot and the chromatic aberrations? I don't want to plot them, but using the information as text.

I could also code it on myself, but would be way easier if there is already something, that I can use and didnt found. 

Thanks for the answer!
I like this package a lot!

**Comment by mjhoptics** [comment, marked as answer]
Hello @gniliac1,
There are several functions in the trace module for calculating the sagittal and tangential astigmatism values. [trace_astigmatism](https://ray-optics.readthedocs.io/en/stable/api/rayoptics.raytr.trace.html#rayoptics.raytr.trace.trace_astigmatism) can be called for a single field point; the function [trace_astigmatism_curve](https://ray-optics.readthedocs.io/en/stable/api/rayoptics.raytr.trace.html#rayoptics.raytr.trace.trace_astigmatism_curve) will sample the optics across the field of view. I haven't written a function to generate distortion curves but it would be patterned after the `trace_astigmatism_curve` function.

For your other requests, you can either generate the data using the classes in [raytr.analyses](https://ray-optics.readthedocs.io/en/stable/api/rayoptics.raytr.analyses.html#module-rayoptics.raytr.analyses) module, such as [RayFan](https://ray-optics.readthedocs.io/en/stable/api/rayoptics.raytr.analyses.html#rayoptics.raytr.analyses.RayFan) and [RayList](https://ray-optics.readthedocs.io/en/stable/api/rayoptics.raytr.analyses.html#rayoptics.raytr.analyses.RayList), or by using the data in the various [mpl](https://ray-optics.readthedocs.io/en/stable/api/rayoptics.mpl.html#module-rayoptics.mpl) plotting options. [RayGeoPSF](https://ray-optics.readthedocs.io/en/stable/api/rayoptics.mpl.analysisfigure.html#rayoptics.mpl.analysisfigure.RayGeoPSF) is a class that takes a `RayList` as input and produces either a spot diagram or 2D histogram; this might be useful to look at to see how the analysis classes are used.

I have a jupyter notebook repo that has a couple of notebooks using these functions. You might look at:
[triplet focus tilt notebook](https://github.com/mjhoptics/ray-optics-notebooks/blob/43dd60e5c693f77d55e4127f5d530397d2666e12/triplet%20focus%20tilt%20notebook.ipynb)
[wavefront notebook](https://github.com/mjhoptics/ray-optics-notebooks/blob/43dd60e5c693f77d55e4127f5d530397d2666e12/wavefront%20notebook.ipynb)
for examples of how to access the data.

Hope this helps.

Mike Hayford
